docs/Developers/Git_Basics.rst

```diff
@@ -283,6 +283,58 @@ another editor will be fired up with all the messages of commits which you want
 to squash. Clean up all the messages and add a new message to be
 displayed for the single commit.
 
+Common Git Issues
+-----------------
+
+Sometimes, you use git add-A and add files you didn't want to your push (often
+after rebasing) and push it to the remote. Here ,is a short outline of, how can
+you remove (or revert changes in) particular files from your commit even after
+pushing to remote.
+
+In your local repo, to revert the file to the state before the previous commit
+run the following:
+::
+
+    $ git checkout HEAD^ /path/to/file
+
+Now , after reverting the file(s) update your last commit , by running :
+::
+
+    $ git commit --amend
+
+To apply these changes to the remote you need to force update the branch :
+::
+
+    $ git push -f myfork
+
+.. note::
+
+    The upstated procedure helps roll back changes by one commit only.
+    Besides, myfork is your forked repository, where you push your commits.
+
+Besides, the ``git checkout`` command offers you more flexibility in reverting
+the changes in a file, done even from earlier than the last commit. By replacing
+the ``HEAD^`` by the revision number of the particular HEAD commit, you can
+refer to the required revision of the file.
+
+Might sound a little intimidating, but don't worry, an example has been
+provided for you.
+First you can check the commit's revision number, where the file was revised by
+running the following command:
+::
+
+    $ git log /path/to/file
+
+The revision number might look like ``3cdc61015724f9965575ba954c8cd4232c8b42e4``
+Now, to revert the file to that commit(or revision) (or just use changes rolled
```

Comment:
Hahaha .... I'll change it though .... Btw the newcomers can choose any one of them too.. :stuck_out_tongue: